### ISSUE TITLE
Missing charge session in Grafana states timelines

### ISSUE THREAD

**Issue by pmboothby** [adriankumpf/teslamate#865]
**Describe the bug**

This morning I did an AC charge in one location, then a DC charge at a second location. In the States, Overview and Trip pages the 2nd (DC) charge is not displayed in the timeline graph. However, both charges display correctly in other visualisations.

**Expected behavior**

Both charges should display in the timeline graphs.

**How to reproduce it (as minimally and precisely as possible)**:

1. Charge the car using AC input.
2. Drive to a DC charger.
3. Charge the car using DC input.
4.  Look at States, Overview and Trip pages in Grafana

**Relevant entries from the logs**

docker-compose logs grafana produces no output while displaying those pages.

**Screenshots**
![Charges](https://user-images.githubusercontent.com/34880264/89747144-bcaa9400-db00-11ea-958c-21dd8d7819be.PNG)
![DC-charge](https://user-images.githubusercontent.com/34880264/89747156-ca601980-db00-11ea-83aa-86c0e14b1f07.PNG)
![AC-charge](https://user-images.githubusercontent.com/34880264/89747151-c633fc00-db00-11ea-958e-7198fafec34d.PNG)
![states](https://user-images.githubusercontent.com/34880264/89747161-d055fa80-db00-11ea-9f57-13809a444c16.PNG)
![overview](https://user-images.githubusercontent.com/34880264/89747163-d350eb00-db00-11ea-815c-2f8e535872fe.PNG)
The pie chart is correct with DC and AC charging percentages & the barchart is includes both, but the timeline is missing the DC charge.
![trip](https://user-images.githubusercontent.com/34880264/89747165-d51aae80-db00-11ea-8d3e-3e74d0dfbcbb.PNG)

**Data**

[charges_csv.txt](https://github.com/adriankumpf/teslamate/files/5048802/charges_csv.txt)
[charging_processes_csv.txt](https://github.com/adriankumpf/teslamate/files/5048852/charging_processes_csv.txt)
[states_csv.txt](https://github.com/adriankumpf/teslamate/files/5048828/states_csv.txt)

**Environment**

- TeslaMate version: 1.19.4
- Type of installation: Docker 
- OS TeslaMate is installed on: Debian

> pboothby@teslamate:/etc$ cat os-release
> PRETTY_NAME="Debian GNU/Linux 10 (buster)"
> NAME="Debian GNU/Linux"
> VERSION_ID="10"
> VERSION="10 (buster)"
> VERSION_CODENAME=buster
> ID=debian
> HOME_URL="https://www.debian.org/"
> SUPPORT_URL="https://www.debian.org/support"
> BUG_REPORT_URL="https://bugs.debian.org/"

Browser: Firefox 79.0 on Windows 7 Pro
- Others:

**Comment by adriankumpf** [comment]
Thanks for the detailed bug report!

We've had a similar bug before, where two events had an identical timestamp and therefore the start date of one event overlapped the other. I suspect that this is happening again

**Comment by pmboothby** [comment]
I'll try to debug it in the next few days if no one else gets to it.

**Comment by pmboothby** [comment]
I've debugged this from the Grafana perspective. The cause of the graph not showing the charging period is because it starts at _exactly_ the same time as the drive ends. This is the relevant data from the Grafana query:
`2020-08-10 09:21:41 2 <- Start of AC charge in Kingscliff`
`2020-08-10 09:38:56 0 <- End of AC charge in Kingscliff`
`2020-08-10 09:45:54 1 <- Start of drive Kingscliff-Byron Bay`
`2020-08-10 10:22:45 0 <- End of drive Kingscliff-Byron Bay`
`2020-08-10 10:22:45 2 <- Start of DC charge in Byron Bay`
`2020-08-10 10:59:55 0 <- End of DC charge in Byron Bay`
`2020-08-10 11:00:44 1 <- Start of drive home from charger`
`2020-08-10 11:07:42 0 <- End of drive home from charger`

Note the rows @ 10:22:45. Strange that the drive seems to be terminated by the start of charging, as if the data stream from the car missing stopping and going into Park. Because the Grafana query sorts the records ascending by state, the graph paints the colour for "Online" instead of "Charging" because it seems to pick the first record to select the colour. I manually updated the data to make the 2nd charge session start at 10:22:46 and the graph displays correctly. 

I'm not clear on where this needs to be fixed...
